.buildkite/dev-docker/run.sh

```diff
@@ -32,28 +32,54 @@ else
 fi
 
 DOCKER_USERNAME=$(vault read -field username "${VAULT_PATH}")
-retry 5 docker login -u ${DOCKER_USERNAME} -p ${DOCKER_PASSWORD} ${DOCKER_USERNAME}
+retry 5 docker login -u ${DOCKER_USERNAME} -p ${DOCKER_PASSWORD} ${DOCKER_REGISTRY}
 unset DOCKER_PASSWORD
 unset DOCKER_USERNAME
 
-tmp_dir=$(mktemp --directory)
-pushd "$tmp_dir"
-git clone https://github.com/elastic/rally
-pushd rally
-# checkout the version from the buildkite branch, but build it from the branch we specified
-if [[ ! -z "${BUILDKITE_BRANCH}" ]]; then
-    git checkout "${BUILDKITE_BRANCH}"
-else
-    git checkout "${BUILD_FROM_BRANCH}"
-fi
-git --no-pager show
+build_docker_image() {
+    tmp_dir=$(mktemp --directory)
+    pushd "$tmp_dir"
+    git clone https://github.com/elastic/rally
+    pushd rally
+    # checkout the version from the buildkite branch, but build it from the branch we specified
+    if [[ ! -z "${BUILDKITE_BRANCH}" ]]; then
+        git checkout "${BUILDKITE_BRANCH}"
+    else
+        git checkout "${BUILD_FROM_BRANCH}"
+    fi
+    git --no-pager show
```

Comment:
I like the idea behind `git --no-pager show`, but this is potentially quite verbose. How about changing this to something a bit more brief, e.g. `git log --oneline -n1`? I would also add comments corresponding to the context where we checkout:

```
echo "Dockerfile commit: $(git log --oneline -n1)"
```

vs.

```
echo "Rally commit: $(git log --oneline -n1)"
```

Reply:
This is a great idea. I'll do that and merge. Thanks!